src/components/CommentCard.jsx

```diff
@@ -1,17 +1,24 @@
 import "../../src/App.css";
 
-const CommentCard = ({ comment }) => {
+const CommentCard = ({ comment, onDeleteComment, isCommentPosted, loggedInUserId }) => {
   const formattedDate = new Date(comment.created_at).toLocaleString();
 
+  const handleDelete = () => {
+    onDeleteComment(comment.comment_id);
+  };
+
   return (
-    <>
-      <section key={comment.comment_id} className="comment-card">
-        <p>{comment.body}</p>
-        <p>
-          Posted by {comment.author} on {formattedDate}
-        </p>
-      </section>
-    </>
+    <section key={comment.comment_id} className="comment-card">
+      <p>{comment.body}</p>
+      <p>
+        Posted by {comment.author} on {formattedDate}
+      </p>
+      {comment.author === loggedInUserId && (
+        <button onClick={handleDelete} disabled={isCommentPosted}>
+          Delete
+        </button>
```

Comment:
Please see my comment below and make sure to avoid multiple click (and errors) by disabling the button while deleting.